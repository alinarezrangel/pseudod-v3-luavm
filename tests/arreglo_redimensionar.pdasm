PDVM 1.0
PLATFORM "pdcrt"

SECTION "code"
  LOCAL 0
  ICONST 1
  ICONST 2
  ICONST 3
  ICONST 4
  MKARR 4
  LSET 0

  -- Escribe el arreglo antes de modificarlo
  LGET 0
  MSG 0, 0, 1
  PRN
  NL

  -- Redimensiónalo: quita elementos
  LGET 0
  ICONST 2
  ROT 1
  MSG 1, 1, 0
  LGET 0
  MSG 0, 0, 1
  PRN
  NL

  -- Redimensiónalo: agrega elementos
  LGET 0
  ICONST 5
  ROT 1
  MSG 1, 1, 0
  LGET 0
  MSG 0, 0, 1
  PRN
  NL

  -- Redimensiónalo: agrega elementos.
  --
  -- Esta vez agregamos muchos elementos para probar la condición `nuevo-tamaño
  -- > capacidad`.
  LGET 0
  ICONST 40
  ROT 1
  MSG 1, 1, 0
  LGET 0
  MSG 0, 0, 1
  PRN
  NL
ENDSECTION

SECTION "procedures"
ENDSECTION

SECTION "constant pool"
  #0 STRING "comoTexto"
  #1 STRING "redimensionar"
ENDSECTION
