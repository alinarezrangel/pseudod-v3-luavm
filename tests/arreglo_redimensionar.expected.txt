(Arreglo#crearCon: 1, 2, 3, 4)
(Arreglo#crearCon: 1, 2)
(Arreglo#crearCon: 1, 2, NULO, NULO, NULO)
(Arreglo#crearCon: 1, 2, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO, NULO)
